osmtm/templates/project.edit.mako

```diff
@@ -36,6 +36,43 @@
   </div>
   </div>
 </div>
+
+<!-- Modal -->
+<div class="modal fade" id="invalidateAllModal" tabindex="-1" role="dialog" aria-labelledby="invalidateAll">
+  <div class="modal-dialog" role="document">
+    <div class="modal-content">
+      <div class="modal-header">
+        <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+        <h4 class="modal-title" id="myModalLabel">${_('Invalidate All')}</h4>
+      </div>
+      <div class="modal-body">
+        <p>
+          ${_("This will mark all tasks currently marked as 'done' as invalid. Please use this only if you are sure of what you are doing.")}
+        </p>
+        <div class="text-center">
+          <div class="form-group">
+            <textarea id="project_invalidate_comment" name="invalidate_all_comment" class="form-control" placeholder="${_('Leave a comment')}" rows="2"></textarea>
```

Comment:
I had to read the code to understand what the comment field was used for. It wasn't obvious to me. I thought that it was a way to ensure that the project manager did really want to do the invalidation. I was tempted to fill it with random text.
I think that this is important for the user to understand that this comment will appear in the history of each task concerned by the invalidation. We can change the placeholder, add a help text below the comment textarea or even write a default comment.